test/inductor/test_binary_folding.py

```diff
@@ -0,0 +1,185 @@
+importlib.import_module("filelock")
+
+from torch.testing._internal.inductor_utils import HAS_CPU, HAS_CUDA
+
+aten = torch.ops.aten
+
+
+class BinaryFoldingTemplate(TestCase):
+    def test_conv_binary_folding(self):
+        @torch.no_grad()
+        def test_conv_fusion(use_bias, module, op, scalar, add_tensor, expect_success):
+            class ConvOp(nn.Module):
+                __constants__ = ["use_scalar"]
+
+                def __init__(self, in_channels, out_channels, device, **kwargs):
+                    super(ConvOp, self).__init__()
+                    self.conv = module(
+                        in_channels, out_channels, bias=use_bias, **kwargs
+                    ).to(device)
+                    self.conv2 = module(
+                        in_channels, out_channels, bias=use_bias, **kwargs
+                    ).to(device)
+                    self.use_scalar = scalar
+                    tensor_size = [1 for _ in range(self.conv.weight.ndim)]
+                    tensor_size[1] = self.conv.weight.size(0)
+                    self.tensor = (
+                        add_tensor
+                        if add_tensor is not None
+                        else torch.rand(tensor_size).to(device)
+                    )
+                    self.op = op
+
+                def forward(self, x):
+                    x = self.conv(x)
+                    if self.use_scalar:
+                        return self.op(x, 2.0)
+                    else:
+                        return self.op(x, self.tensor)
+
+            from torch._inductor.compile_fx import compile_fx, compile_fx_inner
+
+            aten_binary = {
+                torch.add: aten.add.Tensor,
+                torch.sub: aten.sub.Tensor,
+                torch.mul: aten.mul.Tensor,
+                torch.div: aten.div.Tensor,
+            }
+            n_binary_ops = 0
+
+            def my_inner_compile(gm, example_inputs, *args, **kwargs):
+                out = compile_fx_inner(gm, example_inputs, *args, **kwargs)
+                nonlocal n_binary_ops
+                binarry_ops = [n for n in gm.graph.nodes if n.target == aten_binary[op]]
+                n_binary_ops += len(binarry_ops)
+                return out
+
+            torch._dynamo.reset()
+            mod_eager = ConvOp(3, 32, self.device, kernel_size=3, stride=2).eval()
+            out_optimized = torch.compile(
+                mod_eager,
+                backend=functools.partial(compile_fx, inner_compile=my_inner_compile),
+            )
+
+            inps = [4, 3, 4]
+            if module == nn.Conv2d:
+                inps.append(inps[-1])
+            if module == nn.Conv3d:
+                inps.append(inps[-1])
+                inps.append(inps[-1])
+
+            inp = torch.rand(inps).to(self.device)
+            out_eager = mod_eager(inp)
+            out_optimized = out_optimized(inp)
+            self.assertEqual(out_optimized, out_eager, atol=2e-04, rtol=1e-5)
```

Comment:
CPU path works fine even using default tolerance, but for the GPU path, there seems has a big variance, but my local machine(A10) can't reproduce it, I need to find another machine to reproduce it.

Reply:
Skip the pass which cudnn is enabled, it can be passed when disable cudnn convolution.